tasks/db_deploy/migrations/migrate_versions_5_to_6/migrate_sql.py

```diff
@@ -113,3 +113,69 @@ def add_files_storage_class_id_column_sql() -> text:  # pragma: no cover
             FOREIGN KEY (storage_class_id) REFERENCES storage_class (id);
     """
     )
+
+
+def add_mismatch_storage_class_columns_sql() -> text:  # pragma: no cover
+    """
+    SQL for adding the orca_storage_class_id and s3_storage_class columns to the reconcile_catalog_mismatch_report table.
+    New cells will contain '1', the id for GLACIER.
+
+    Returns: SQL for adding the column.
+    """
+    return text(
+        """
+        -- Add the storage_class columns with a default of 1/GLACIER
+        ALTER TABLE reconcile_catalog_mismatch_report
+        ADD COLUMN IF NOT EXISTS orca_storage_class_id int2 NOT NULL default 1,
+        ADD COLUMN IF NOT EXISTS s3_storage_class text NOT NULL default 'GLACIER';
+        COMMENT ON COLUMN reconcile_catalog_mismatch_report.orca_storage_class_id
+            IS 'Storage class of the file as reported in the ORCA catalog.';
+        COMMENT ON COLUMN reconcile_catalog_mismatch_report.s3_storage_class
+            IS 'Storage class of the file as reported in the S3 bucket.';
+
+        -- Remove the default now that new cells are populated.
+        ALTER TABLE reconcile_catalog_mismatch_report
+        ALTER COLUMN orca_storage_class_id DROP DEFAULT,
+        ALTER COLUMN s3_storage_class DROP DEFAULT;
+        
+        -- Drop the FK constraint since ADD CONSTRAINT IF NOT EXISTS isn't an option
+        ALTER TABLE reconcile_catalog_mismatch_report
+            DROP CONSTRAINT IF EXISTS FK_mismatch_orca_storage_class;
+        
+        -- Add the FK constraint
+        ALTER TABLE reconcile_catalog_mismatch_report
+        ADD CONSTRAINT FK_mismatch_orca_storage_class
+            FOREIGN KEY (orca_storage_class_id) REFERENCES storage_class (id);
+    """
+    )
+
```

Comment:
Do you feel we should add some SQL to cleanup the reports? So if the reports in mismatch were due to what we preciously checked was lastupdate, should we remove those records? Or do we just let it be since the reports are basically short lived and provide migrators information that the reports  may now be out of date and to only worry about mismatch reports after the upgrade?

Reply:
We caught that before release, so customers won't have any last-update mismatches. https://github.com/nasa/cumulus-orca/commit/7f9529207988dd9d2b33eb1c0e26c4bec85c7cee